scenarios/kubernetes_build.py

```diff
@@ -30,8 +30,40 @@ def check(*cmd):
     print >>sys.stderr, 'Run:', cmd
     subprocess.check_call(cmd)
 
+def check_output(*cmd):
+    """Log and run the command, raising on errors, return output"""
+    print >>sys.stderr, 'Run:', cmd
+    return subprocess.check_output(cmd)
+
+def check_build_exists(gcs):
+    """ check if a k8s/federation build with same version
+        already exists in remote path
+    """
+    version = ''
+    try:
+        check('bazel', '--batch', 'build', '//:save_git_version')
+        if os.path.exists('bazel-genfiles/version'):
+            with open('bazel-genfiles/version', 'r') as fp:
+                version = fp.read()
+    except subprocess.CalledProcessError as exc:
+        # fallback with doing a real build
+        print >>sys.stderr, 'Failed to get k8s version, continue: %s' % exc
+        return False
+
+    if version:
+        if not gcs:
+            gcs = 'gs://kubernetes-release-dev'
+        gcs = os.path.join(gcs, 'ci', version)
```

Comment:
need to add the suffix (if set) to `ci` too.

Reply:
done